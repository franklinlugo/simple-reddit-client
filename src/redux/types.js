export const FETCH_DATA = 'FETCH_DATA';
export const FETCH_DATA_SUCCESS = 'FETCH_DATA_SUCCESS';
export const FETCH_DATA_ERROR = 'FETCH_DATA_ERROR';

export const REMOVE_POST_BY_ID = 'REMOVE_POST_BY_ID';
export const SET_ACTIVE_POST = 'SET_ACTIVE_POST';
