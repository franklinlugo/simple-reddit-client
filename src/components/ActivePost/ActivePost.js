import React from 'react';
import { useSelector } from 'react-redux';
import { Container, Title, Image } from './ActivePost.styles';

function ActivePost() {
  const post = useSelector((state) => state.activePost);
  const imageSrc = post?.preview?.images[0].resolutions[0].url;

  return (
    <Container>
      {post && Object.keys(post).length && (
        <>
          <Title>{post.title}</Title>
          <Image src={imageSrc} />
        </>
      )}
    </Container>
  );
}

export default ActivePost;
