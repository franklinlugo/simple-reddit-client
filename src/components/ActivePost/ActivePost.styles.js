import styled from 'styled-components';

export const Container = styled.div`
  flex: 1;
  position: fixed;
  top: 0;
  width: calc(100% - 500px);
  height: 100%;
  margin-left: 500px;
`;

export const Title = styled.h3`
  margin: 1rem;
  padding: 0;
  line-height: 1;
  text-align: center;
  color: ${(props) => props.theme.darkgray};
`;

export const Image = styled.img`
  display: block;
  margin: 0 auto;
`;
