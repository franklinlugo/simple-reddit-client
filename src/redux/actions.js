import * as TYPES from './types';
import { getTopPost } from '../services';

export const onFetchStart = () => ({ type: TYPES.FETCH_DATA });

export const onFetchSuccess = (payload) => ({
  type: TYPES.FETCH_DATA_SUCCESS,
  payload,
});

export const onFetchError = () => ({ type: TYPES.FETCH_DATA_ERROR });

export const onFetch = () => async (dispatch) => {
  dispatch(onFetchStart());
  try {
    const posts = await getTopPost();
    dispatch(onFetchSuccess(posts));
  } catch (error) {
    dispatch(onFetchError());
  }
};
