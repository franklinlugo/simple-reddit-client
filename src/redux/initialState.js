export const initialState = {
  loading: null,
  error: null,
  posts: [],
};
