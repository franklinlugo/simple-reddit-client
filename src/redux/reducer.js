import { initialState } from './initialState';
import * as TYPES from './types';

export function reducer(state = initialState, action) {
  switch (action.type) {
    case TYPES.FETCH_DATA: {
      return {
        ...state,
        loading: true,
        error: null,
      };
    }
    case TYPES.FETCH_DATA_SUCCESS: {
      return {
        ...state,
        loading: false,
        posts: action.payload,
      };
    }
    case TYPES.FETCH_DATA_ERROR: {
      return {
        ...state,
        loading: false,
        error: true,
      };
    }
    case TYPES.REMOVE_POST_BY_ID: {
      const id = action.payload;
      return {
        ...state,
        posts: state.posts.filter((post) => post.id !== id),
      };
    }
    default:
      return state;
  }
}
