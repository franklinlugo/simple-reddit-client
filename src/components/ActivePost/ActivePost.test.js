import React from 'react';
import ActivePost from './ActivePost';
import { renderWithProvider } from '../../utils/tests/renderWithProvider';
import { postsList } from '../../utils/tests/fixtures';
import { setActivePost } from '../../redux/actions';

test('ActivePost should render', async () => {
  const { asFragment } = renderWithProvider(<ActivePost />, {
    onPrepareStore: (store) => store.dispatch(setActivePost(postsList[0])),
  });
  expect(asFragment()).toMatchSnapshot();
});
