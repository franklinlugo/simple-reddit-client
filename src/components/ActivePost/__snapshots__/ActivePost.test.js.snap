// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivePost should render 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: fixed;
  top: 0;
  width: calc(100% - 500px);
  height: 100%;
  margin-left: 500px;
}

.c1 {
  margin: 1rem;
  padding: 0;
  line-height: 1;
  text-align: center;
}

.c2 {
  display: block;
  margin: 0 auto;
}

<div
    class="c0"
  >
    <h3
      class="c1"
    >
      111 title
    </h3>
    <img
      class="c2"
      src="https://image1111.com"
    />
  </div>
</DocumentFragment>
`;
